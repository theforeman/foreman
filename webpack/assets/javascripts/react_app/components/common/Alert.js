import React from 'react';
import Icon from './Icon';
import { ALERT_CSS } from '../../constants';

const Alert = (props) => {
  let classNames = ALERT_CSS[props.type];

  if (props.css) {
    classNames += ' ' + props.css;
  }

  if (props.dismissable) {
    classNames += ' ' + ALERT_CSS.dismissable;
  }

  const closeBtn = (
    <button className="close" data-dismiss="alert" aria-hidden="true">
      <Icon type="close" />
    </button>
  );

  return (
    <div className={classNames}>
      {props.dismissable && closeBtn}
      {props.children}
    </div>
  );
};

Alert.defaultProps = { dismissable: false, css: null, type: 'info' };
export default Alert;
