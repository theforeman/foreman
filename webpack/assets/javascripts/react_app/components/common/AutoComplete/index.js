import React, { Component } from 'react';
import Downshift from 'downshift';
import { Dropdown, MenuItem, InputGroup, FormControl, Button } from 'patternfly-react';
import PropTypes from 'prop-types';

export const renderItems = ({
  items, highlightedIndex, selectedItem, getItemProps,
}) => (
  <Dropdown.Menu
    style={{
      display: 'block',
      position: 'absolute',
      left: 'inherit',
      marginTop: 0,
      top: 'auto',
    }}
  >
    {items.map((item, index) => (
      <MenuItem
        {...getItemProps({
          index,
          item,
          active: highlightedIndex === index,
          onClick: (e) => {
            // At this point the event.defaultPrevented
            // is already set to true by react-bootstrap
            // MenuItem. We need to set it back to false
            // So downshift will execute it's own handler
            e.defaultPrevented = false;
          },
        })}
        key={item}
      >
        {item}
      </MenuItem>
    ))}
  </Dropdown.Menu>
);

class AutoComplete extends Component {
  constructor(props) {
    super(props);

    this.state = {
      inputValue: '',
    };
  }

  render() {
    const {
      onSearch, labelText, onInputUpdate, items, actionText, ...rest
    } = this.props;

    return (
      <Downshift
        onStateChange={({ inputValue }) => {
          if (typeof inputValue === 'string') {
            onInputUpdate(inputValue);
            this.setState({ inputValue });
          }
        }}
        defaultHighlightedIndex={0}
        selectedItem={this.state.inputValue}
        {...rest}
      >
        {({
          getInputProps,
          getItemProps,
          getLabelProps,
          isOpen,
          inputValue,
          highlightedIndex,
          getRootProps,
          selectedItem,
        }) => (
          <div>
            {labelText && <label {...getLabelProps()}>{labelText}</label>}
            <InputGroup>
              <FormControl type="text" {...getInputProps()} />
              <InputGroup.Button>
                <Button onClick={() => onSearch(inputValue)}>{actionText}</Button>
              </InputGroup.Button>
            </InputGroup>

            {isOpen && items.length
              ? renderItems({
                  items,
                  highlightedIndex,
                  selectedItem,
                  getItemProps,
                })
              : null}
          </div>
        )}
      </Downshift>
    );
  }
}

AutoComplete.propTypes = {
  items: PropTypes.arrayOf(PropTypes.string).isRequired,
  onInputUpdate: PropTypes.func.isRequired,
  onSearch: PropTypes.func.isRequired,
  labelText: PropTypes.string,
  actionText: PropTypes.string,
};

AutoComplete.defaultProps = {
  labelText: '',
  actionText: 'Search',
};

export default AutoComplete;
