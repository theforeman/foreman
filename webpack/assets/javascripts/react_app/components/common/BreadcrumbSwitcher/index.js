import React from 'react';
import {
  Popover,
  Button,
  Icon,
  ListGroup,
  ListGroupItem,
  OverlayTrigger,
} from 'patternfly-react';
import { ButtonToolbar } from 'react-bootstrap';
import EllipsisWithTooltip from 'react-ellipsis-with-tooltip';
import PropTypes from 'prop-types';
import './BreadcrumbSwitcher.scss';

const BreadcrumbSwitcher = ({ resources }) => {
  const popover = (
    <Popover id="popover-contained">
      <ListGroup className="scrollable-list">
        {resources.map((item) => {
          const { onClick, url, caption } = item;
          return (
            <ListGroupItem
              className="no-border"
              key={`id-${caption}`}
              href={url}
              onClick={onClick}
            >
              <EllipsisWithTooltip>{caption}</EllipsisWithTooltip>
            </ListGroupItem>
          );
        })}
      </ListGroup>
    </Popover>
  );
  return (
    <ButtonToolbar>
      <OverlayTrigger
        rootClose
        trigger="click"
        placement="bottom"
        overlay={popover}
      >
        <Button>
          <Icon type="fa" name="exchange" />
        </Button>
      </OverlayTrigger>
    </ButtonToolbar>
  );
};

BreadcrumbSwitcher.propTypes = {
  href: PropTypes.string,
  onClick: PropTypes.func,
  caption: PropTypes.string,
};

export default BreadcrumbSwitcher;
