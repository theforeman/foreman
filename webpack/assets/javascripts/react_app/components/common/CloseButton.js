import React from 'react';
import Icon from './Icon';

const CloseButton = ({onClick}) => {
  return (
    <button className="close" aria-hidden="true" onClick={onClick}>
      <Icon type="close" />
    </button>
  );
};

CloseButton.propTypes = {
  onClick: React.PropTypes.func.isRequired
};

export default CloseButton;
