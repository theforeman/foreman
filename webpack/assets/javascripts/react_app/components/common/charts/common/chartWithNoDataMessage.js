import React from 'react';
import MessageBox from '../../MessageBox';

export const chartWithNoDataMessage = (WrappedComp, noDataMsg = 'No data available') => (props) => {
  const columns = props && props.data && props.data.columns ? props.data.columns : [];
  if (columns.length > 0) {
    return (<WrappedComp {...props} />);
  }
  return <MessageBox msg={noDataMsg} icontype="info" />;
};
