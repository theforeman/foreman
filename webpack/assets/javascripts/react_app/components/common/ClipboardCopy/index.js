import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Button, OverlayTrigger, Tooltip } from 'patternfly-react';
import UUID from 'uuid/v1';
import { translate as __ } from '../../../common/I18n';
import './clipboard-copy.scss';

const ClipboardCopy = ({
  text: defaultText,
  successMessage,
  buttonText,
  textareaProps,
  buttonProps,
}) => {
  const [text, setText] = useState(defaultText);

  return (
    <div className="clipboard-copy">
      <textarea
        defaultValue={text}
        onChange={({ target: { value } }) => setText(value)}
        {...textareaProps}
      />
      <OverlayTrigger
        overlay={<Tooltip id={UUID()}>{successMessage}</Tooltip>}
        placement="right"
        trigger={['click']}
        rootClose
      >
        <Button
          onClick={() => navigator.clipboard.writeText(text)}
          bsStyle="default"
          {...buttonProps}
        >
          {buttonText}
        </Button>
      </OverlayTrigger>
    </div>
  );
};

ClipboardCopy.propTypes = {
  text: PropTypes.string.isRequired,
  buttonText: PropTypes.string,
  successMessage: PropTypes.string,
  textareaProps: PropTypes.object,
  buttonProps: PropTypes.object,
};

ClipboardCopy.defaultProps = {
  buttonText: __('Copy to clipboard'),
  successMessage: __('Copied!'),
  textareaProps: {},
  buttonProps: {},
};

export default ClipboardCopy;
