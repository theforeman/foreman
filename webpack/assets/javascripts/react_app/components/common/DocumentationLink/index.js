import React from 'react';
import PropTypes from 'prop-types';
import { MenuItem, Icon } from 'patternfly-react';
import { newWindowOnClick } from '../../../common/helpers';
import { translate as __ } from '../../../../react_app/common/I18n';

const DocumentationLink = ({ href, children }) => (
  <MenuItem key="documentationUrl" href={href} onClick={newWindowOnClick(href)}>
    <Icon type="fa" name="question-circle" />
    {` ${children}`}
  </MenuItem>
);

DocumentationLink.propTypes = {
  href: PropTypes.string.isRequired,
  children: PropTypes.node,
};

DocumentationLink.defaultProps = {
  children: __('Documentation'),
};

export default DocumentationLink;
