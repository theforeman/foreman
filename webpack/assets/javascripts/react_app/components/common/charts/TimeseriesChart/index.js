import React from 'react';
import { LineChart as PfLineChart } from 'patternfly-react';
import { chartWithNoDataMessage } from '../common/chartWithNoDataMessage';
import { getTimeseriesChartConfig } from '../../../../../services/ChartService';

const TimeseriesChartWithNoDataMsgBox = chartWithNoDataMessage(PfLineChart);

const TimeseriesChart = ({
  data,
  type = 'line',
  unloadData = false,
  getConfig = getTimeseriesChartConfig,
}) => <TimeseriesChartWithNoDataMsgBox
       {...getConfig({ data, type })}
       unloadBeforeLoad = {unloadData}
      />;

export default TimeseriesChart;
