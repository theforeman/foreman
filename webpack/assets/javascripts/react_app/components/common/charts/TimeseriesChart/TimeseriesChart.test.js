import TimeseriesChart from './';
import { timeseriesChartConfig } from '../../../../../services/ChartService.consts';
import { testComponentSnapshotsWithFixtures } from '../../../../common/testHelpers';

jest.unmock('./');
const fixtures = {
  'render timeseries chart': {
    data: null,
    getConfig: () => Object.assign({}, {
      ...timeseriesChartConfig,
      data: {
        x: 'time',
        columns: [['time', 15029999123], ['Runtime', 8], ['Config Retrieval', 35]],
      },
    }),
  },
  'render empty timeseries chart': {
    data: null,
    getConfig: () => timeseriesChartConfig,
  },
};
testComponentSnapshotsWithFixtures(TimeseriesChart, fixtures);
