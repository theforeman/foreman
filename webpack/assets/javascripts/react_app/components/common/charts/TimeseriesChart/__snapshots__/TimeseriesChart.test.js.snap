// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render empty timeseries chart 1`] = `
<Component
  axis={
    Object {
      "x": Object {
        "tick": Object {
          "fit": false,
        },
        "type": "timeseries",
      },
    }
  }
  data={
    Object {
      "columns": Array [],
      "x": "time",
    }
  }
  subchart={
    Object {
      "show": true,
    }
  }
  tooltip={
    Object {
      "format": Object {
        "title": [Function],
      },
      "grouped": false,
    }
  }
  unloadBeforeLoad={false}
  zoom={
    Object {
      "enabled": true,
    }
  }
/>
`;

exports[`render timeseries chart 1`] = `
<Component
  axis={
    Object {
      "x": Object {
        "tick": Object {
          "fit": false,
        },
        "type": "timeseries",
      },
    }
  }
  data={
    Object {
      "columns": Array [
        Array [
          "time",
          15029999123,
        ],
        Array [
          "Runtime",
          8,
        ],
        Array [
          "Config Retrieval",
          35,
        ],
      ],
      "x": "time",
    }
  }
  subchart={
    Object {
      "show": true,
    }
  }
  tooltip={
    Object {
      "format": Object {
        "title": [Function],
      },
      "grouped": false,
    }
  }
  unloadBeforeLoad={false}
  zoom={
    Object {
      "enabled": true,
    }
  }
/>
`;
